src/rust/lib/graph-editor/src/lib.rs

```diff
@@ -1573,50 +1636,103 @@ fn new_graph_editor(world:&World) -> GraphEditor {
     }
 
 
-    // === Remove Edge ===
-    frp::extend! { network
-
-    rm_input_edges       <- any (inputs.remove_all_node_edges, inputs.remove_all_node_input_edges);
-    rm_output_edges      <- any (inputs.remove_all_node_edges, inputs.remove_all_node_output_edges);
-    input_edges_to_rm    <= rm_input_edges  . map(f!((node_id) model.node_in_edges(node_id)));
-    output_edges_to_rm   <= rm_output_edges . map(f!((node_id) model.node_out_edges(node_id)));
-    edges_to_rm          <- any (inputs.remove_edge, input_edges_to_rm, output_edges_to_rm);
-    outputs.edge_removed <+ edges_to_rm;
-    }
-
-
     // === Set Node Expression ===
     frp::extend! { network
 
     outputs.node_expression_set <+ inputs.set_node_expression;
 
 
+
+    // ==================
     // === Move Nodes ===
+    // ==================
+
+    mouse_pos_fix <- mouse.position.map(|p| Vector2(p.x,p.y));
+
+
+    // === Discovering drag targets ===
+
+    let main            = touch.nodes.down.clone_ref();
+    let main_pressed    = touch.nodes.is_down.clone_ref();
+    main_was_sel       <- main.map(f!((id) model.nodes.selected.contains(id)));
+    tgts_if_non_sel    <- main.map(|id|vec![*id]).gate_not(&main_was_sel);
+    tgts_if_sel        <- main.map(f_!(model.nodes.selected.items())).gate(&main_was_sel);
+    tgts               <- any(tgts_if_non_sel,tgts_if_sel);
+    eval tgts ((ids) model.disable_grid_snapping_for(ids));
+
+    main_pos_on_press       <- main.map(f!((id) model.node_position(id)));
+    mouse_pos_on_press      <- mouse_pos_fix.sample(&main);
+    mouse_pos_diff          <- mouse_pos_fix.map2(&mouse_pos_on_press,|t,s|t-s).gate(&main_pressed);
+    main_tgt_pos_rt_changed <- mouse_pos_diff.map2(&main_pos_on_press,|t,s|t+s);
+    just_pressed            <- bool (&main_tgt_pos_rt_changed,&main_pos_on_press);
+    main_tgt_pos_rt         <- any  (&main_tgt_pos_rt_changed,&main_pos_on_press);
+
+
+    // === Snapping ===
+
+    let main_tgt_pos_anim = Animation::<Vector2<f32>>::new(&network);
+    let x_snap_strength     = Tween::new(&network);
```

Comment:
Can these two snap strength components (and their further usage) be sensibly grouped into Vector2?

Reply:
nope, the movement will be different.